### PR TITLE
Structured reasons for rejection

### PR DESCRIPTION
**Pull request by steventux**
## Context

We need to capture specific information about why a candidate's application is being rejected so that the feedback given is relevant and useful to the candidate.

## Changes proposed in this pull request

Adds structured questions/reasons for rejecting an application in the provider interface and persists them in the same structure. 

### Initial questions with validation fail
image

### Other reasons for rejection

image

### Checking / Confirming answers
image

## Guidance to review

The bulk of the complexity is in the conditional validation logic, I've tried to keep this as simple as I can but there's a lot of it.

Does the feature flag strategy make sense in terms of releasing this feature and switching from the simpler single rejection reason?

## Link to Trello card

https://trello.com/c/azXp4WN4/2238-%F0%9F%8F%88-structured-reasons-for-rejection

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/controllers/provider_interface/reasons_for_rejection_controller.rb [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+    attr_accessor :interested_in_future_applications_y_n
+
+    attr_accessor :why_are_you_rejecting_this_application
+
+    def to_prose
+      safeguarding_y_n.to_s
+    end
+  end
+
+  class ReasonsForRejectionWizard
```
Comment (AgaDufrat) [inline app/controllers/provider_interface/reasons_for_rejection_controller.rb]: Would it be worth moving out of the controller to eg `app/forms/provider_interface`? 
Similarly the `ReasonsForRejection` class.

File: config/locales/reasons_for_rejection.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      no_degree: 'No degree'
+  activemodel:
+    errors:
+      models:
+        provider_interface/reasons_for_rejection_wizard:
+          attributes:
+            candidate_behaviour_y_n:
+              blank: Select yes if it was due to candidate behaviour
+            candidate_behaviour_what_did_the_candidate_do:
+              blank: Select at least one reason related to the candidate's behaviour
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:19]:
```suggestion
              blank: Select at least one reason related to the candidate’s behaviour
```

File: config/locales/reasons_for_rejection.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+              blank: Select yes if it was because this course is full
+            offered_on_another_course_y_n:
+              blank: Select yes if it was because you offered them a place on another course
+            offered_on_another_course_details:
+              blank: Enter details about the place offered on another course
+              too_long: The details you give about an offer on another course must be 100 words or fewer
+            honesty_and_professionalism_y_n:
+              blank: Select yes if it was due to concerns about their honesty and professionalism
+            honesty_and_professionalism_concerns:
+              blank: Select at least one reason related to the candidate's honesty and professionalism
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:64]:
```suggestion
              blank: Select at least one reason related to the candidate’s honesty and professionalism
```

File: config/locales/reasons_for_rejection.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+              blank: Enter details about the information they disclosed making them unsuitable to work with children
+              too_long: The details you give about the information they disclosed must be 100 words or fewer
+            safeguarding_concerns_vetting_disclosed_information_details:
+              blank: Enter details about the information revealed by your vetting process
+              too_long: The details you give about the information revealed by your vetting process must be 100 words or fewer
+            safeguarding_concerns_other_details:
+              blank: Enter details about safeguarding
+              too_long: The details you give about safeguarding must be 100 words or fewer
+            why_are_you_rejecting_this_application:
+              blank: Enter details about why you're rejecting their application
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:91]:
```suggestion
              blank: Enter details about why you’re rejecting their application
```

File: config/locales/reasons_for_rejection.yml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give for rejecting this application must be 200 words or fewer
+            other_advice_or_feedback_y_n:
+              blank: Select yes if there is any other advice or feedback you‘d like to give
+            other_advice_or_feedback_details:
+              blank: Enter any other advice or feedback
+              too_long: The advice or feedback you give must be 200 words or fewer
+            interested_in_future_applications_y_n:
+              blank: Select yes if you'd be interested in future applications from this candidate
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:99]:
```suggestion
              blank: Select yes if you’d be interested in future applications from this candidate
```

**Review by adamsilver — DISMISSED** [review]

**Comment by adamsilver** [comment]
@steventux the feedback is showing up buggy and the content isn’t quite right:

image

Should be like this:

image

**Comment by duncanjbrown** [comment]
@adamsilver most thorough design review in a PR ever? ⭐

File: spec/system/provider_interface/reject_an_application_spec.rb [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    expect(page).to have_content('Additional advice')
+    expect(page).to have_content('While impressive, your parkour skills are not relevant')
+  end
+
+  def and_i_submit_the_reasons_for_rejection
+    click_on 'Reject application'
+  end
+
+  def then_i_can_see_the_reasons_were_saved
+    expect(page).to have_content('Application rejected')
```
Comment (mnacos) [inline spec/system/provider_interface/reject_an_application_spec.rb:149]: I imagine this step will include some more checks around the display of structured rejection reasons?

**Review by mnacos — COMMENTED** [review]
Haven't fully reviewed this yet, but it seems to me we'll need a component for displaying structured reasons on application 'show'. Not sure yet if this should be re-used on the 'check step', because the 'show' component will also need to cope with unstructured reasons.

All this may be tackled in a different story, of course, as we have the feature flag.

File: spec/system/provider_interface/reject_an_application_spec.rb (hunk 6 above)
Comment (steventux) [reply spec/system/provider_interface/reject_an_application_spec.rb:149]: @mnacos see https://github.com/DFE-Digital/apply-for-teacher-training/pull/3404/commits/4ca7143c010b08116efbc151bfd5e56083a35c45 for component to display reasons for rejection

Comment (steventux) [reply spec/system/provider_interface/reject_an_application_spec.rb:149]: @adamsilver https://github.com/DFE-Digital/apply-for-teacher-training/commit/27396ca292b495ce36c024e37b4c28464d64a0a7 should fix the inset text problem when displaying reasons for rejection.

File: config/locales/reasons_for_rejection.yml [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+      no_english_gcse: 'No English GCSE grade 4 (C) or above, or valid equivalent'
+      no_science_gcse: 'No Science GCSE grade 4 (C) or above, or valid equivalent (for primary applicants)'
+      no_degree: 'No degree'
+  activemodel:
+    errors:
+      models:
+        provider_interface/reasons_for_rejection_wizard:
+          attributes:
+            candidate_behaviour_y_n:
+              blank: Select yes if it was due to candidate behaviour
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:17]:
```suggestion
              blank: Select yes if it was related to candidate behaviour
```

File: config/locales/reasons_for_rejection.yml [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+            candidate_behaviour_what_did_the_candidate_do:
+              blank: Select at least one reason related to the candidate's behaviour
+            candidate_behaviour_other:
+              blank: Enter details about what the candidate did
+              too_long: The details you give about their behaviour must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:22]:
```suggestion
              too_long: Details about candidate behaviour must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    errors:
+      models:
+        provider_interface/reasons_for_rejection_wizard:
+          attributes:
+            candidate_behaviour_y_n:
+              blank: Select yes if it was due to candidate behaviour
+            candidate_behaviour_what_did_the_candidate_do:
+              blank: Select at least one reason related to the candidate's behaviour
+            candidate_behaviour_other:
+              blank: Enter details about what the candidate did
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:21]:
```suggestion
              blank: Enter details about candidate behaviour
```

File: config/locales/reasons_for_rejection.yml [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give about their behaviour must be 100 words or fewer
+            candidate_behaviour_what_to_improve:
+              blank: Enter details about how the candidate can improve their behaviour
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:24]:
```suggestion
              blank: Enter details about how candidate can improve their behaviour
```

File: config/locales/reasons_for_rejection.yml [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give about how the candidate can improve their behaviour must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:25]:
```suggestion
              too_long: Details about how the candidate can improve their behaviour must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+            quality_of_application_y_n:
+              blank: Select yes if it was due to the quality of their application
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:27]:
```suggestion
              blank: Select yes if it was related to the quality of their application
```

File: config/locales/reasons_for_rejection.yml [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+            quality_of_application_which_parts_needed_improvement:
+              blank: Select at least one reason related to the quality of their application
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:29]:
```suggestion
              blank: Select at least one reason related to the quality of the candidate’s application
```

File: config/locales/reasons_for_rejection.yml [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+            quality_of_application_personal_statement_what_to_improve:
+              blank: Enter details about how they can improve their personal statement
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:31]:
```suggestion
              blank: Enter details about how the candidate can improve their personal statement
```

File: config/locales/reasons_for_rejection.yml [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give about how they can improve their personal statement must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:32]:
```suggestion
              too_long: Details about how they can improve their personal statement must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+            quality_of_application_subject_knowledge_what_to_improve:
+              blank: Enter details about how they can improve their subject knowledge
+              too_long: The details you give about how the candidate can improve their subject knowledge must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:35]:
```suggestion
              too_long: Details about how the candidate can improve their subject knowledge must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+            quality_of_application_other_details:
+              blank: Enter details about the quality of the their application
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:37]:
```suggestion
              blank: Enter details about the quality of their application
```

File: config/locales/reasons_for_rejection.yml [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give about the quality of their application must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:38]:
```suggestion
              too_long: Details about the quality of their application must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+            quality_of_application_other_what_to_improve:
+              blank: Enter how they can improve their application
+              too_long: The details you give about how they can improve their application must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:41]:
```suggestion
              too_long: Details about how they can improve their application must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 20]
```diff
@@ -0,0 +1,99 @@
...
+              blank: Enter details about how they can improve their personal statement
+              too_long: The details you give about how they can improve their personal statement must be 100 words or fewer
+            quality_of_application_subject_knowledge_what_to_improve:
+              blank: Enter details about how they can improve their subject knowledge
+              too_long: The details you give about how the candidate can improve their subject knowledge must be 100 words or fewer
+            quality_of_application_other_details:
+              blank: Enter details about the quality of the their application
+              too_long: The details you give about the quality of their application must be 100 words or fewer
+            quality_of_application_other_what_to_improve:
+              blank: Enter how they can improve their application
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:40]:
```suggestion
              blank: Enter how they can improve the quality their application
```

File: config/locales/reasons_for_rejection.yml [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give about how they can improve their application must be 100 words or fewer
+            qualifications_y_n:
+              blank: Select yes if it was due to their qualifications
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:43]:
```suggestion
              blank: Select yes if it was related to their qualifications
```

File: config/locales/reasons_for_rejection.yml [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+            qualifications_which_qualifications:
+              blank: Select at least one reason related to their qualifications
+            qualifications_other_details:
+              blank: Enter details about the issue with their qualifications
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:47]:
```suggestion
              blank: Enter details about the problem with their qualifications
```

File: config/locales/reasons_for_rejection.yml [hunk 23]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give about their qualifications must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:48]:
```suggestion
              too_long: Details about their qualifications must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 24]
```diff
@@ -0,0 +1,99 @@
...
+            performance_at_interview_y_n:
+              blank: Select yes if it was related to their performance at interview
+            performance_at_interview_what_to_improve:
+              blank: Enter details about how they can improve their performance at interview
+              too_long: The details you give about how they can improve their performance at interview must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:53]:
```suggestion
              too_long: Details about how they can improve their performance at interview must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 25]
```diff
@@ -0,0 +1,99 @@
...
+            course_full_y_n:
+              blank: Select yes if it was because this course is full
+            offered_on_another_course_y_n:
+              blank: Select yes if it was because you offered them a place on another course
+            offered_on_another_course_details:
+              blank: Enter details about the place offered on another course
+              too_long: The details you give about an offer on another course must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:60]:
```suggestion
              too_long: Details about the place offered on another course must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 26]
```diff
@@ -0,0 +1,99 @@
...
+            honesty_and_professionalism_y_n:
+              blank: Select yes if it was due to concerns about their honesty and professionalism
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:62]:
```suggestion
              blank: Select yes if it was related to concerns about their honesty and professionalism
```

File: config/locales/reasons_for_rejection.yml [hunk 27]
```diff
@@ -0,0 +1,99 @@
...
+            honesty_and_professionalism_concerns:
+              blank: Select at least one reason related to the candidate's honesty and professionalism
+            honesty_and_professionalism_concerns_information_false_or_inaccurate_details:
+              blank: Enter details related to false or inaccurate information given
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:66]:
```suggestion
              blank: Enter details about the inaccurate or false information in the application
```

File: config/locales/reasons_for_rejection.yml [hunk 28]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give related to false or inaccurate information given must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:67]:
```suggestion
              too_long: Details about inaccurate or false information in the application must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 29]
```diff
@@ -0,0 +1,99 @@
...
+            honesty_and_professionalism_concerns_plagiarism_details:
+              blank: Enter details related to evidence of plagiarism in personal statement or elsewhere
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:69]:
```suggestion
              blank: Enter details related to evidence of plagiarism in the application
```

File: config/locales/reasons_for_rejection.yml [hunk 30]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give related to evidence of plagiarism in personal statement or elsewhere must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:70]:
```suggestion
              too_long: The details about evidence of plagiarism in the application must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 31]
```diff
@@ -0,0 +1,99 @@
...
+            honesty_and_professionalism_concerns_references_details:
+              blank: Enter details related to references not supporting the application
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:72]:
```suggestion
              blank: Enter details about references not supporting the application
```

File: config/locales/reasons_for_rejection.yml [hunk 32]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give related to references must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:73]:
```suggestion
              too_long: Details about references not supporting the application must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 33]
```diff
@@ -0,0 +1,99 @@
...
+            honesty_and_professionalism_concerns_other_details:
+              blank: Enter details about the candidate’s honesty and professionalism
+              too_long: The details you give related to their honesty and professionalism must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:76]:
```suggestion
              too_long: Details about the candidate’s honesty and professionalism must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 34]
```diff
@@ -0,0 +1,99 @@
...
+            safeguarding_y_n:
+              blank: Select yes if it was related to safeguarding
+            safeguarding_concerns:
+              blank: Select at least one reason related to safeguarding
+            safeguarding_concerns_candidate_disclosed_information_details:
+              blank: Enter details about the information they disclosed making them unsuitable to work with children
+              too_long: The details you give about the information they disclosed must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:83]:
```suggestion
              too_long: Details about the information the candidate disclosed must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 35]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give related to references must be 100 words or fewer
+            honesty_and_professionalism_concerns_other_details:
+              blank: Enter details about the candidate’s honesty and professionalism
+              too_long: The details you give related to their honesty and professionalism must be 100 words or fewer
+            safeguarding_y_n:
+              blank: Select yes if it was related to safeguarding
+            safeguarding_concerns:
+              blank: Select at least one reason related to safeguarding
+            safeguarding_concerns_candidate_disclosed_information_details:
+              blank: Enter details about the information they disclosed making them unsuitable to work with children
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:82]:
```suggestion
              blank: Enter details about information the candidate disclosed which makes them unsuitable to work with children
```

File: config/locales/reasons_for_rejection.yml [hunk 36]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give about the information they disclosed must be 100 words or fewer
+            safeguarding_concerns_vetting_disclosed_information_details:
+              blank: Enter details about the information revealed by your vetting process
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:85]:
```suggestion
              blank: Enter details about the information found by your vetting process which makes the candidate unsuitable to work with children
```

File: config/locales/reasons_for_rejection.yml [hunk 37]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The details you give about the information revealed by your vetting process must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:86]:
```suggestion
              too_long: Details about the information found by your vetting process which makes the candidate unsuitable to work with children must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 38]
```diff
@@ -0,0 +1,99 @@
...
+            safeguarding_concerns_other_details:
+              blank: Enter details about safeguarding
+              too_long: The details you give about safeguarding must be 100 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:89]:
```suggestion
              too_long: Details about safeguarding must be 100 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 39]
```diff
@@ -0,0 +1,99 @@
...
+            why_are_you_rejecting_this_application:
+              blank: Enter details about why you're rejecting their application
+              too_long: The details you give for rejecting this application must be 200 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:92]:
```suggestion
              too_long: Details about why you’re rejecting this application must be 200 words or fewer
```

File: config/locales/reasons_for_rejection.yml [hunk 40]
```diff
@@ -0,0 +1,99 @@
...
+            other_advice_or_feedback_y_n:
+              blank: Select yes if there is any other advice or feedback you‘d like to give
+            other_advice_or_feedback_details:
+              blank: Enter any other advice or feedback
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:96]:
```suggestion
              blank: Enter any other advice or feedback
```

File: config/locales/reasons_for_rejection.yml [hunk 41]
```diff
@@ -0,0 +1,99 @@
...
+              too_long: The advice or feedback you give must be 200 words or fewer
```
Comment (adamsilver) [inline config/locales/reasons_for_rejection.yml:97]:
```suggestion
              too_long: The other advice or feedback must be 200 words or fewer
```

File: app/views/provider_interface/reasons_for_rejection/edit_initial_questions.html.erb [hunk 42]
```diff
@@ -0,0 +1,185 @@
...
+
+              <%= f.govuk_text_area :quality_of_application_other_details, label: { text: 'Please give details' }, max_words: 100 %>
+              <%= f.govuk_text_area :quality_of_application_other_what_to_improve, label: { text: 'What could they do to improve?' }, max_words: 100 %>
+            <% end %>
+          <% end %>
+        <% end %>
+        <%= f.govuk_radio_button :quality_of_application_y_n, 'No', label: { text: 'No' } %>
+      <% end %>
+
+      <%= f.govuk_radio_buttons_fieldset(:qualifications_y_n, legend: { size: 'm', text: 'Was it related to qualifications?' }) do %>
```
Comment (adamsilver) [inline app/views/provider_interface/reasons_for_rejection/edit_initial_questions.html.erb:68]:
```suggestion
      <%= f.govuk_radio_buttons_fieldset(:qualifications_y_n, legend: { size: 'm', text: 'Was it related to their qualifications?' }) do %>
```

**Comment by mnacos** [comment]
On the review app, I'm stuck on the commit page:

image

**Comment by mnacos** [comment]
Also, attempting to view a newly rejected application, (with safeguarding concenrs from the vetting process), I get a 500.

image

File: app/components/provider_interface/reasons_for_rejection_component.html.erb [hunk 43]
```diff
@@ -0,0 +1,193 @@
...
+        </p>
+      <% end %>
+    </div>
+  <% end %>
+
+  <% if reasons_for_rejection.other_advice_or_feedback_y_n == 'Yes' %>
+    <div class="app-feedback">
+      <%= content_tag(subheading_tag_name, 'Additional advice', class: 'govuk-heading-s') %>
+      <p class="govuk-body"><%= reasons_for_rejection.other_advice_or_feedback_details %></p>
+      <% if editable? %>
```
Comment (mnacos) [inline app/components/provider_interface/reasons_for_rejection_component.html.erb:186]: :+1:

File: app/components/provider_interface/status_box_components/rejected_component.html.erb [hunk 44]
```diff
@@ -1,3 +1,13 @@
 <h2 class="govuk-heading-m govuk-!-font-size-27">Rejection details</h2>
 
-<%= render SummaryListComponent.new(rows: rejected_rows) %>
+<% if @application_choice.structured_rejection_reasons.present? %>
```
Comment (mnacos) [inline app/components/provider_interface/status_box_components/rejected_component.html.erb:3]: Great

**Review by mnacos — COMMENTED** [review]
Thanks for including the presentation component within the application choice #show. I've flagged some weirdness on the review app (it may turn out to be the review app). Otherwise good.

**Comment by duncanjbrown** [comment]
So nearly there! I checked "Would like them to apply again" but didn't see the "Future applications" section in the summary. And in the prototype the checkboxes are now big — is that correct @adamsilver ?

**Comment by duncanjbrown** [comment]
Two more qs: 

- If I select "no" for either "safeguarding issues" or "honesty and professionalism" the go back and select "yes" for "safeguarding issues" should the summary still contain the "any other advice" field?
- If I click "cancel" on the last page should my answers so far be cleared?

**Comment by steventux** [comment]
@duncanjbrown I'd assume the answer to your first question is no, if the step flow omits the 2nd step because of a change to the answers on the first then the summary shouldn't contain these. Another fun quirk of clearing out answers, @adamsilver can you confirm?
Second point: In the prototype cancel takes you back to the application but answers are retained, is this correct @adamsilver ?

**Comment by steventux** [comment]
@duncanjbrown https://github.com/DFE-Digital/apply-for-teacher-training/pull/3404/commits/b67e0a275105af6e52d4052da3c9a5b7b1dbdee8 unsets other reasons if conditions change and these become irrelevant as per your question here https://github.com/DFE-Digital/apply-for-teacher-training/pull/3404#issuecomment-735712177 I'm assuming this is the right thing to do. I've addressed check box size and added the missing _Future applications_ section to the summary in the two commits before this.

**Review by duncanjbrown — APPROVED** [review]
LGTM!!! 🚀

**Pull request merged**